Documentation/Yarn-Spec.md

```diff
@@ -0,0 +1,636 @@
+
+Despite this being the first version of the Yarn language to ever exist it is version 2 of the Yarn language.
+This is because despite Yarn and Yarn Spinner being separate elements attempting to not stay, at least initially, in version lock with the de facto implementation is just asking for trouble, hence version 2 of Yarn.
+
+### Coverage
+
+This document covers the elements of a Yarn script that are required to be considered a valid yarn script as well as the rules necessary for an implementing program to be able to conform to this spec.
+
+An *implementing program* is a program that accepts yarn files as inputs and understands them based on the rules in this document.
+This document does not cover how a yarn file is to be transformed or handled internally by an implementing program.
+
+This document does note when behaviours are *unspecified*.
+The implementing program may choose how to handle any unspecified behaviour, often unspecified behaviours have no good solution and should result in an error to the user.
+An example of unspecified behaviour are required tags on Nodes.
+Only the `Title` tag is required, an implementation may choose to make other tags required or banned.
+
+### Assumptions
+
+There is one large assumption in this document and about the language itself.
+That is it that it is intended to be embedded into a larger game project.
+
+As such details like how lines are views, how selections are made, how directions get used, or how functions are handled are all assumed to be passed off onto the larger game to handle.
+This is not technically necessary but does explain what would otherwise be gaps in the language.
+If your project does not have a game to offload certain tasks to these elements will have to be handled by the implementing program itself.
+
+### Reading This Specification
+
+`monofont` terms are to be taken as literals.
+As an example the `Title` tag would be written as a string literal `"Title"` in C#.
+
+*italics* terms are terms which are reused throughout the document.
+They are presented in italics the first time they are defined.
+
+_Must_ is a hard requirement.
+_Should_ is a recommendation, albeit a strong one.
+
+Dates, spelling, and numbers are to be in Australian English.
+
+### Modifying This Specification
+
+Once this specification is complete it is set and unchanging.
+Modifications beyond clarifications won't be allowed.
+To make changes to the language specification a new version of the language will need to be created.
+
+This is to ensure implementing programs can conform to a language version and not have to worry about it changing after the fact.
+
+## File Format
+
+Yarn files must be a UTF-8 text file without BOM set.
+File extension should be `.yarn`.
+
+### Lines
+
+The *line* is the common unit that comprises all elements in a yarn file.
+A line is a series of characters terminated by the *new line* symbol.
+The new line symbol should be the `\n` character.
```

Comment:
Windows Notepad outputs `\r\n` and cannot be changed. Considering that Windows Notepad will be used in many cases for manual editing, it would be prudent to allow `\r\n` (this could be as simple as stipulating `\r` characters should be ignored and considered zero-width whitespace)

Reply:
hmm I get what you are saying here, I still think it should be `\n` by default recommendation but the wording should be changed to say "The new line symbol should be the \n character however any combination of characters may be used as the newline symbol." or something to that effect that allows for `\r\n`, or any exotic combination of symbols so desired.

With that said, really should be recommending people *not* be using notepad, notepad does weird things to files. The number of times I've had students come to me with borked code and it turns out notepad did it is far beyond count.